python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_base_group_chat.py

```diff
@@ -356,13 +363,22 @@ async def main() -> None:
         """
 
         # Create the first chat message if the task is a string or a chat message.
-        first_chat_message: ChatMessage | None = None
+        first_chat_message: ChatMessage | List[ChatMessage] | None = None
         if task is None:
             pass
         elif isinstance(task, str):
             first_chat_message = TextMessage(content=task, source="user")
-        elif isinstance(task, TextMessage | MultiModalMessage | StopMessage | HandoffMessage):
+        elif isinstance(task, get_args(ChatMessage)[0]):
             first_chat_message = task
+        elif isinstance(task, list):
+            if not task:
+                raise ValueError("Task list cannot be empty")
+            if not all(isinstance(msg, get_args(ChatMessage)[0]) for msg in task):
+                raise ValueError("All messages in task list must be valid ChatMessage types")
+            first_chat_message = task[0]
+            # Queue remaining messages for processing
+            for msg in task[1:]:
+                await self._output_message_queue.put(msg)
```

Comment:
Since we are already emitting messages from the group chat manager, we should just emit all the messages there rather than from the group chat run method.

Reply:
Thanks for the suggestion, implementing passing all the messages at once; also renamed first_chat_message to messages